src/anomalib/metrics/spro.py

```diff
@@ -0,0 +1,158 @@
+        super().__init__(**kwargs)
+        self.threshold = threshold
+        self.saturation_config = saturation_config
+        self.add_state("preds", default=[], dist_reduce_fx="cat")
+        self.add_state("targets", default=[], dist_reduce_fx="cat")
+
+    def update(self, predictions: torch.Tensor, _: torch.Tensor, masks: torch.Tensor) -> None:
+        """Compute the SPRO score for the current batch.
+
+        Args:
+            predictions (torch.Tensor): Predicted anomaly masks
+            _ (torch.Tensor): Unused argument, but needed for different metrics within the same AnomalibMetricCollection
+            masks (torch.Tensor): Ground truth anomaly masks with non-binary values and, original height and width
+
+        Example:
+            To update the metric state for the current batch, use the ``update`` method:
+
+            >>> spro.update(preds, _, labels)
+        """
+        assert masks is not None
+        self.targets.append(masks)
+        self.preds.append(predictions)
+
+    def compute(self) -> torch.Tensor:
+        """Compute the macro average of the SPRO score across all masks in all batches.
+
+        Example:
+            To compute the metric based on the state accumulated from multiple batches, use the ``compute`` method:
+
+            >>> spro.compute()
+            tensor(0.5433)
+        """
+        targets = dim_zero_cat(self.targets)
+        preds = dim_zero_cat(self.preds)
+        return spro_score(preds, targets, threshold=self.threshold, saturation_config=self.saturation_config)
+
+
+def spro_score(
+    predictions: torch.Tensor,
+    targets: torch.Tensor,
+    threshold: float = 0.5,
+    saturation_config: dict | None = None,
+) -> torch.Tensor:
+    """Calculate the SPRO score for a batch of predictions.
+
+    Args:
+        predictions (torch.Tensor): Predicted anomaly masks
+        targets: (torch.Tensor): Ground truth anomaly masks with non-binary values and, original height and width
+        threshold (float): When predictions are passed as float, the threshold is used to binarize the predictions.
+        saturation_config (dict): Saturations configuration for each label (pixel value) as the keys.
+            Defaults: ``None`` (which the score is equivalent to PRO metric, but with the 'region' are
+            separated by mask files.
+
+    Returns:
+        torch.Tensor: Scalar value representing the average SPRO score for the input batch.
+    """
+    predictions = torch.nn.functional.interpolate(predictions.unsqueeze(1), targets.shape[1:])
+
+    # Apply threshold to binary predictions
+    if predictions.dtype == torch.float:
+        predictions = predictions > threshold
+
+    score = torch.tensor(0.0)
+    m = 0
+    # Iterate for each image in the batch
+    for i, target in enumerate(targets):
+        unique_labels = torch.unique(target)
+
+        # Iterate for each ground-truth mask per image
+        for label in unique_labels[1:]:
```

Comment:
can you expand the comment, explaining why starting from unique_labels[1:]

Reply:
Sure, I will add more comment, explaining the index 0 is the background/normal part of the masks

Reply:
The latest implementation is not using `unique_labels[1:]` anymore. But i have provided the related comment in this line:

https://github.com/openvinotoolkit/anomalib/blob/3bbc7507ea29f537118728b579e72e04b998f78f/src/anomalib/metrics/spro.py#L136